### PR TITLE
Make compatible with GHC 8.8

### PR DESCRIPTION
**Pull request by wwwater**

### CODE REVIEW

**Comment by wwwater** [comment]
One needs to update `network-arbitrary` first too, for tests to compile.

File: network-uri-json.cabal [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name:                network-uri-json
-version:             0.3.1.1
+version:             0.3.1.2
```
Comment (alunduil) [inline network-uri-json.cabal:2]: I use git flow to manage these versions so I'll bump this version when I cut the release based on the included changes.  You can go ahead and remove this from this diff.  Once this passes tests and lands I'll create the release branch.

**Comment by alunduil** [comment]
I wonder why the travis results no longer show up in these diffs.  I'll probably have to get haskell-ci working and ensure the travis configuration is up to date as well.

File: network-uri-json.cabal (hunk 1 above)
Comment (wwwater) [reply network-uri-json.cabal]: Done!

Comment (alunduil) [reply network-uri-json.cabal]: Thanks!

**Comment by alunduil** [comment]
I've got someone who's attempting to get network-arbitrary fixed up so we can ensure the tests continue passing here.  If you don't mind, I'd like to wait for that release before landing this to keep everything green.   I don't think any other changes would be needed here other than potentially updating the version on network-arbitrary when it's ready.

**Comment by alunduil** [comment]
Actually, one thing I forgot, can you ensure that the GHC tested-with version in the cabal file is updated to include GHC 8.8?

**Comment by alunduil** [comment]
You should be able to increase the network-arbitrary version to 0.6.0.0 and get this passing travis.  I'll land this as soon as it passes.

File: network-uri-json.cabal [hunk 2]
```diff
@@ -15,7 +15,7 @@ copyright:           (c) 2017 Alex Brandt
 category:            Network
 build-type:          Simple
 cabal-version:       >=1.10
-tested-with:         GHC >= 7.6 && < 8.2.1 || > 8.2.1 && < 9.0
+tested-with:         GHC >= 7.6 && < 8.8.3
```
Comment (wwwater) [inline network-uri-json.cabal:18]: I've updated this, but I didn't understand the syntax it had before, why was the second condition after `||`? It wasn't correct.

**Comment by wwwater** [comment]
> You should be able to increase the network-arbitrary version to 0.6.0.0 and get this passing travis. I'll land this as soon as it passes.

Done

**Comment by wwwater** [comment]
Not sure why travis fails for GHC `7.8.1` and for `8.6.1`. Maybe it would make sense to add `8.6.5` or `8.8.3` to travis too.

File: network-uri-json.cabal (hunk 2 above)
Comment (alunduil) [reply network-uri-json.cabal:18]: I was excluding version 8.2.1 but I don't recall why.  I'd say leave the exclusion in and don't specify the minor version in the gop GHC version.  Looking at it, the previous version looks like what I'd go with and we probably just need to run:

```
haskell-ci --hlint regenerate network-uri-json.cabal
```

If you don't mind adding that, I'll see about the travis failures (I expect some version mismatches with network-arbitrary but we'll see).

**Comment by alunduil** [comment]
The travis error looks like it's due to an old dependency.  Once you run haskell-ci and update the travis configuration, we should see things that are impacted by these libraries only.

File: network-uri-json.cabal (hunk 2 above)
Comment (wwwater) [reply network-uri-json.cabal:18]: well, then `GHC >= 7.6 && < 8.2.1 || > 8.2.1 && < 9.0` was alright, I'll revert back to it

**Comment by wwwater** [comment]
Now it's identical to #32 :D

**Comment by alunduil** [comment]
I'll merge this one once we have the travis checks passing.  In this case, the failing version is due to a bad dependency resolution.  We can use the same trick as before and just drop that version since it's not listed in the hackage tested version (see https://matrix.hackage.haskell.org/#/package/network-uri-json).

Let me know if that sounds good otherwise, we'll have to do some magic with cabal conditionals on the dependency versions (which I'd rather avoid since every other case works).

**Pull request merged**